### ISSUE TITLE
Portable Mode

### ISSUE THREAD

**Issue by smaragdus** [murrty/youtube-dl-gui#26]
[youtube-dl-gui](https://github.com/murrty/youtube-dl-gui) saves its configuration file (**user.config**) in AppData:

`C:\Users\User\AppData\Local\youtube_dl_gui\youtube-dl-gui.exe_Url_axgmtcpdpseegko5bud4bxc054vq5g0j\2.0.0.0\user.config`

I would like to suggest an option configuration file to be saved in program folder (next to executable  file- **youtube-dl-gui.exe**) which will mean that **youtube-dl-gui** is running in portable mode.

Portable mode can be implemented in many different ways, I will suggest just few:

1. **command-line argument** - if the executable (**youtube-dl-gui.exe**) is started with a specific parameter, for example `youtube-dl-gui.exe -portable`, configuration file (**user.config**) is saved in program folder;
2. **configuration file** - when **youtube-dl-gui** starts it checks program folder for configuration file (**user.config**) and if it is there the program uses it;
3. **option in settings** - new check box- **Portable mode** (perhaps in **General** tab), which, if selected, forces the program to save configuration file (**user.config**) in program folder;
4. **portable flag** - specific file in program folder (for example **portable** without extension or **portable.txt**) triggers portable mode, when **youtube-dl-gui** starts it checks program folder for such file and if it is there configuration file (**user.config**) is saved in program folder;

Any of these methods would be fine for me but I prefer  option **2**- when the program starts it checks its program folder for configuration file (**user.config**).

I prefer  to use programs running in portable mode- they are easier to backup, they can be easily transferred from machine to machine with their settings, they are handier for testing.

I suppose that the implementation of portable mode would not require too much work.

**Comment by murrty** [comment]
I can look into it, yes. The way I implement it, however, may not be pretty, since I don't have the skill set required to actually redirect the settings provider at this time.

**Comment by smaragdus** [comment]
@murrty 

I am not a programmer so I cannot help with  the code. However I suppose that it might be useful for you to have a look at another [youtube-dl](https://github.com/ytdl-org/youtube-dl) open source front-end also written in C#/.NET- **Vividl**:

- [GitHub](https://github.com/Bluegrams/Vividl)
- [SourceForge](https://sourceforge.net/projects/vividl/)
- [Download](https://sourceforge.net/projects/vividl/files/)

which has a portable version that saves configuration files (`portable.config` and `Vividl.exe.config`) in program folder. Perhaps looking at the source code may help you to find a way to implement an option to save [youtube-dl-gui:](https://github.com/murrty/youtube-dl-gui) configuration file (`user.config`) in program folder.

Regards

**Comment by murrty** [comment]
Hello,

I decided that I'll just stick with a, surprise surprise, ini file. An outdated, but still reliable method of saving settings (if you don't have equal signs in your strings). As of 8cfae26, it's available to anyone with `settings.ini` in the same folder as youtube-dl-gui. I plan on adding a button to convert current settings to portable and vice-versa, but I'm not gonna worry about it right now. I'm going to consider this finished since it's essentially implemented.

**Comment by smaragdus** [comment]
@murrty 

Hey, what a good news! I love outdated methods and outdated software! :)

I turned towards portable software (I use such software whenever possible) many years ago when my newly bought laptop turned out to had defective internal drives. But before I realized this and subsequently replaced them I had to re-install Windows several times so I had to go through the painful and time-consuming process of installing hundreds of programs. So from then on portability became a matter of principle for me.

For me INI files are great- they can be easily edited manually. Years ago I came upon a portable program (I cannot remember now which it was) which used INI file for settings but strangely it failed to save a particular setting via the program, but when I edited the INI file manually I had no problem, the program read the INI file just right.

I saw that there is a new release- version [2.23](https://github.com/murrty/youtube-dl-gui/releases/tag/2.23), I will just check it out.

Thank you!
